core/src/main/java/bio/terra/pearl/core/service/workflow/TaskDispatcher.java

```diff
@@ -193,11 +192,21 @@ private void assignScheduledTasks(StudyEnvironment studyEnv) {
      * will assign a recurring task to enrollees who have already taken it at least once, but are due to take it again
      */
     private void assignRecurring(T taskConfig) {
-        List<Enrollee> enrollees = enrolleeService.findWithTaskInPast(
-                taskConfig.getStudyEnvironmentId(),
-                taskConfig.getStableId(),
```

Comment:
I tried updating `findWithTaskInPast` but encoding all of the new logic we need into it proved to be frustrating, so I opted for less efficient but most correct solution of loading all enrollees and then just calling `isRecurrenceWindowOpen`